index.js

```diff
@@ -61,9 +67,24 @@ module.exports = {
             testCredentials = {};
         }
 
+        if (typeof testFlow === "string") {
+                testFlow = JSON.parse(testFlow);
+        }
+
+        var flowId = testFlow[0].z || "f1";
+
+        testFlow.forEach(function(node) {
+            node.z = flowId;
+            if (node.type == "debug") {
```

Comment:
Not sure we should be swapping node types.  With this change, this module cannot ever be used to test the debug node, or another node called 'debug'.  Do we want the core code to use this helper for consistency?

Reply:
My thinking was that it would be good to be able to develop test flows in the editor and then export them into the test script without needing to change anything.  As there is no helper node in the editor, the debug node is the logical choice as it would be used for debugging in the editor any way.

In any case, it turns out that the debug node won't load under test helper because parts of Node Red aren't properly initialised.

Reply:
Hmm.  The debug node tests run and pass in the core using the helper script there.  will need to look at why this is happening.

Reply:
The debug node tests definitely don't run under node-red-node-test-helper.  The reason is that Node Red is only being partially initialised by the test helper and functions relating to interacting with the editor haven't been added to the RED object passed to the node when it is created.  This problem will affect any other node that interacts with the editor, such as inject

Reply:
The Debug node tests in core use the core test helper which is virtually identical to this one. So what specifically doesn't work? What errors do you get?

Reply:
Running 58-debug_spec.js under the test helper gives:

      Uncaught TypeError: Cannot set property message of AssertionError which has only a getter
      at test/58-debug_spec.js:88:21                                                                                          
      at WebSocket.<anonymous> (test/58-debug_spec.js:605:13)                                                                 
      at Receiver._receiver.onmessage (/usr/lib/node_modules/ws/lib/websocket.js:137:47)                                      
      at Receiver.dataMessage (/usr/lib/node_modules/ws/lib/receiver.js:409:14)                                               
      at perMessageDeflate.decompress (/usr/lib/node_modules/ws/lib/receiver.js:366:40)                                       
      at _decompress (/usr/lib/node_modules/ws/lib/permessage-deflate.js:309:9)                                               
      at _inflate.flush (/usr/lib/node_modules/ws/lib/permessage-deflate.js:395:7)                                            
      at afterWrite (_stream_writable.js:454:3)                                                                               
      at onwrite (_stream_writable.js:445:7)                                                                                  
      at InflateRaw.afterTransform (_stream_transform.js:90:3)                                                                
      at Zlib.callback (zlib.js:515:5)   

Running any tests that include debug nodes give:

     TypeError: Cannot read property 'post' of null
      at Array.module.exports (/usr/lib/node_modules/node-red/nodes/core/core/58-debug.js:223:19)
      at Object.load (index.js:112:28)
      at Context.<anonymous> (test/lower-case_spec.js:36:12)

Reply:
From the stack trace it looks like you're attempting to test the debug node with the lower case test?

Looks like the core debug node tests call helper.startServer(), but lower-case tests don't.

The way to try it is to see if the core debug tests run with the module.  When I copy the core debug node to the module examples (including the lib folder), and the debug test to the test helper project, the debug node tests pass for me.